### DISCUSSION TITLE
React-Select field undefined when using render

### DISCUSSION THREAD

**Discussion by mmvcode** [react-hook-form/react-hook-form#2235] [Q&A] [answered]
I have a form with a few React-Select dropdowns. The dropdown is a custom one, looking like this:

```
    <Controller
          name={name}
          control={refControl}
          render={(props) => {
            const args = {
              id,
              name,
              options,
              defaultValue,
              isMulti,
              isClearable,
              onChange,
              ...props,
            };

            return isCreatable ? (
              <CreatableSelect {...args} onCreateOption={onCreateOption} />
            ) : (
              <ReactSelect {...args} />
            );
          }}
        />
```

[https://codesandbox.io/s/frosty-violet-6t6fg?file=/src/Dropdown.jsx](https://codesandbox.io/s/frosty-violet-6t6fg?file=/src/Dropdown.jsx)

The problem I got is that, when submitted, the values are `undefined`. Am I missing something here or am I doing something wrong?

Thanks.

**Comment by bluebill1049** [comment, marked as answer]
did you use `onChange` to update the value?

**Reply by mmvcode** [reply]
> 
> 
> did you use `onChange` to update the value?

No. In the code shown above, `onChange` is actually `undefined`, but the `Controller` is passing it

**Reply by bluebill1049** [reply]
that's the problem you should use `props.onChange('you value')` otherwise, RHF will not getting the updated value. hope this fix your problem.

**Reply by mmvcode** [reply]
Yes, that was it. Simply adding `onChange={(v) => props.onChange(v)` fixes the issue.

**Reply by albertolotz** [reply]
mmcCode, How you call a function and get de value at same time?
